imaginary.go

```diff
@@ -38,6 +38,7 @@ var (
 	aCertFile           = flag.String("certfile", "", "TLS certificate file path")
 	aKeyFile            = flag.String("keyfile", "", "TLS private key file path")
 	aAuthorization      = flag.String("authorization", "", "Defines a constant Authorization header value passed to all the image source servers. -enable-url-source flag must be defined. This overwrites authorization headers forwarding behavior via X-Forward-Authorization")
+	aCustomHeaders      = flag.String("custom-headers", "", "Forwards custom headers to the image source server. -enable-url-source flag must be defined.")
```

Comment:
I'm not a fan of the wording here "Custom Headers" doesn't directly convey the intent of what the flag does. Personally I would prefer the flag be named "forward-headers" instead. It also matches your description better.